### ISSUE TITLE
Link phase may fail if headers are found but libraries aren't

### ISSUE THREAD

**Issue by emcrisostomo** [emcrisostomo/fswatch#95]
A FreeBSD 10 user reported link failure while resolving `inotify` functions:

```
CXXLD fswatch
../libfswatch/src/libfswatch/.libs/libfswatch.so: undefined reference to `inotify_add_watch'
../libfswatch/src/libfswatch/.libs/libfswatch.so: undefined reference to `inotify_init'
../libfswatch/src/libfswatch/.libs/libfswatch.so: undefined reference to `inotify_rm_watch'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
*** [fswatch] Error code 1
```

`./configure` was finding the `inotify` header files:

```
checking sys/inotify.h usability... yes
checking sys/inotify.h presence... yes
checking for sys/inotify.h... yes
```

but the functions could not be resolved.

`./configure` was correctly reporting `inotify` headers as present since they were installed by `libinotify`.

The `./configure` scripts should be enhanced so as to check for functions after headers are found.

**Comment by emcrisostomo** [comment]
Resolved for `kqueue` and `inotify` monitor in `develop`.

**Comment by emcrisostomo** [comment]
Closed in `1.6.0-develop`.
